generators/angular/templates/src/main/webapp/app/admin/user-management/user-management.route.ts.ejs

```diff
@@ -27,18 +27,20 @@ import UserManagementDetailComponent from './detail/user-management-detail.compo
 import UserManagementUpdateComponent from './update/user-management-update.component';
 
 @Injectable({ providedIn: 'root' })
-export class UserManagementResolve implements Resolve<IUser | null> {
+class UserManagement {
   constructor(private service: UserManagementService) {}
 
-  resolve(route: ActivatedRouteSnapshot): Observable<IUser | null> {
-    const id = route.params['login'];
+  getUser(id: string): Observable<IUser | null> {
     if (id) {
       return this.service.find(id);
     }
     return of(null);
   }
 }
 
+export const UserManagementResolve: ResolveFn<IUser | null> = (route: ActivatedRouteSnapshot, state: RouterStateSnapshot) =>
+  inject(UserManagement).getUser(route.paramMap.get('login')!);
```

Comment:
`UserManagement` is not necessary and can be dropped.

Casting to ResolveFn was not possible at https://github.com/jhipster/generator-jhipster/pull/22482.
Since is an union of functions, it's complicated to revert to the actual api to use at tests.
We should omit the casting to follow the same pattern.

```suggestion
export const UserManagementResolve = (route: ActivatedRouteSnapshot) => {
  const login = route.paramMap.get('login');
  if (login) {
    return inject(UserManagementService).find(login);
  }
  return of(null);
};
```